source/h5vol/H5VolUtil.c

```diff
@@ -0,0 +1,232 @@
+#include "H5VolUtil.h"
+#include "H5VolError.h"
+
+void *safe_calloc(size_t n, size_t s, unsigned long line)
+{
+    if (n == 0)
+    {
+        return NULL;
+    }
+
+    void *p = calloc(n, s);
+    if (!p)
+    {
+        fprintf(stderr, "[%s:%ld]Out of memory at calloc (%ld, %ld)\n",
+                __FILE__, line, (unsigned long)n, (unsigned long)s);
+        exit(EXIT_FAILURE);
+    }
+    return p;
+}
+
+void *safe_malloc(size_t n, unsigned long line)
+{
+    if (n == 0)
+    {
+        return NULL;
+    }
+
+    void *p = malloc(n);
+    if (!p)
+    {
+        fprintf(stderr, "[%s:%ld]Out of memory at malloc (%ld bytes)\n",
+                __FILE__, line, (unsigned long)n);
+        exit(EXIT_FAILURE);
```

Comment:
FWIW, since C99 one can use `(%zu bytes)` to print a `size_t` argument, then you don't need to cast. (I don't know if you need to support older standards, though.)